src/pages/Swap/index.tsx

```diff
@@ -113,6 +113,19 @@ const SwapPageUi: React.FC<SwapPageUiProps> = ({
         currentFormValues.direction === 'exactAmountIn' ? 'exactAmountOut' : 'exactAmountIn',
     }));
 
+  const maxFromInput = () => {
+    if (!fromTokenUserBalanceWei) {
+      return new BigNumber(0).toFixed();
+    }
+
+    const maxFromInputTokens = convertWeiToTokens({
+      valueWei: fromTokenUserBalanceWei,
+      token: formValues.fromToken,
+    });
+
+    return maxFromInputTokens.toFixed();
+  };
```

Comment:
In order to prevent unnecessary processing, you can memoize this variable instead of using a function:

```ts
const maxFromInput = useMemo(() => {
    if (!fromTokenUserBalanceWei) {
      return new BigNumber(0).toFixed();
    }

    const maxFromInputTokens = convertWeiToTokens({
      valueWei: fromTokenUserBalanceWei,
      token: formValues.fromToken,
    });

    return maxFromInputTokens.toFixed();
  }, [formValues.fromToken, fromTokenUserBalanceWei]);
```